mariner/server.go

```diff
@@ -516,6 +512,33 @@ func (server *Server) authZ(r *http.Request) bool {
 	return authResponse.Auth
 }
 
+// fetchRefreshToken is invoked from the server to check if a refresh token is expired and fetches a new one if it is.
+func (server *Server) fetchRefreshToken() bool {
+	wtsPath := fmt.Sprintf("https://%s/wts/oauth2/", os.Getenv("GEN3_HOSTNAME"))
```

Comment:
There is no need to communicate to the WTS via outside hostname, better use URL like this:

```suggestion
	wtsPath := "http://workspace-token-service/oauth2/"
```